src/main/java/org/jfree/chart/date/SerialDate.java

```diff
@@ -538,14 +552,17 @@ public static SerialDate addDays(int days, SerialDate base) {
      *
      * @return a new date.
      */
+    @SuppressWarnings({"index", "value"}) // True positive, fixed upstream bug
     public static SerialDate addMonths(int months, SerialDate base) {
         int yy = (12 * base.getYYYY() + base.getMonth() + months - 1) / 12;
         if (yy < MINIMUM_YEAR_SUPPORTED || yy > MAXIMUM_YEAR_SUPPORTED) {
             throw new IllegalArgumentException("Call to addMonths resulted in unsupported year");
         }
         int mm = (12 * base.getYYYY() + base.getMonth() + months - 1) % 12 + 1;
-        int dd = Math.min(base.getDayOfMonth(), 
-                SerialDate.lastDayOfMonth(mm, yy));
+        int lastDayOfMonth = SerialDate.lastDayOfMonth(mm, yy);
```

Comment:
What is the point of abstracting out this variable?
(Maybe there used to be a `@SuppressWarnings` on it?)
Please review the diffs and undo any unnecessary changes like this, to minimize the diffs.